draft-ietf-gnap-core-protocol.md

```diff
@@ -2256,7 +2256,7 @@ Subject identifiers returned by the AS SHOULD uniquely identify the RO at the
 AS. Some forms of subject identifier are opaque to the client instance (such as the subject of an 
 issuer and subject pair), while others forms (such as email address and phone number) are
 intended to allow the client instance to correlate the identifier with other account information
-at the client instance. The client instance MUST NOT request or use any returned subject identifiers for communication
+at the client instance. The AS MUST validate the correspondance between the user and the returned subject identifiers. The client instance MUST NOT request or use any returned subject identifiers for communication
```

Comment:
correspondance - typo

"The AS MUST validate" appears to be an important security check, but it's not clear to me what we're expecting the AS to do. It seems to me that the AS, once it has authenticated the RO (a.k.a. end user) simply returns all it knows about the RO. So there's nothing to validate.

Reply:
@yaronf the process you're describing is one way to do the validation in question (arguably the most common way).

Reply:
In that case I would say the proposed language is confusing. How about:

> The AS MUST ensure that the returned subject identifiers only apply to the authenticated end user.

Reply:
Accepted Yaron's suggested text.